### PR TITLE
bt_atomic: use atomic macro to replace wireless/bluetooth/bt_atomic.c

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
bt_atomic: use atomic macro to replace wireless/bluetooth/bt_atomic.c

## Impact
bt

## Testing
ci

### CODE REVIEW

File: wireless/bluetooth/bt_atomic.c [hunk 1]
```diff
@@ -38,10 +44,10 @@ bt_atomic_t bt_atomic_incr(FAR bt_atomic_t *ptr)
   irqstate_t flags;
   bt_atomic_t value;
 
-  flags = spin_lock_irqsave(NULL);
+  flags = spin_lock_irqsave_wo_note(&g_bt_atomic_lock);
```
Comment (anchao) [inline wireless/bluetooth/bt_atomic.c:47]: why not replace to atomic_* api directly

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR description does not adequately meet the NuttX requirements. While it provides *some* information, it's far too brief and lacks crucial details.  Here's a breakdown of what's missing:

* **Summary:**  "bt_atomic: use small lock in wireless/bluetooth/bt_atomic.c" is a title, not a summary.  It *states* what the change is, but not *why* it's necessary, *how* it works, or its potential impact.  There's no mention of related issues.

* **Impact:**  Simply stating "bt" is insufficient.  It needs to specify the impact on users, build process, hardware, documentation, security, and compatibility.  Even if the answer is "NO" for most of these, it should be explicitly stated.

* **Testing:** "ci" suggests it relies on Continuous Integration testing. While CI is important, it's insufficient on its own. The PR needs to provide specific details about the local testing environment (host OS, compiler, target architecture, board) and *show* the testing logs before and after the change, demonstrating the improvement or fix.  Simply stating "ci" doesn't provide evidence of proper testing.

To be acceptable, the PR description needs significant expansion, providing the missing context and details as outlined in the NuttX requirements.

File: wireless/bluetooth/bt_atomic.c (hunk 1 above)
Comment (hujun260) [reply wireless/bluetooth/bt_atomic.c]: done

**Review by anchao — APPROVED** [review]
Emm... I might remove bt_atomic.c and replace all of them with stdatomic's API, but this change is also fine for me.

**Comment by xiaoxiang781216** [comment]
> Emm... I might remove bt_atomic.c and replace all of them with stdatomic's API, but this change is also fine for me.

Yes, it's better to remove bt_atomic.c and change bt_atomic to macro to avoid the function invocation cost.

File: wireless/bluetooth/bt_atomic.c [hunk 2]
```diff
@@ -35,78 +35,57 @@
...
 bt_atomic_t bt_atomic_setbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno)
 {
-  irqstate_t flags;
   bt_atomic_t value;
 
-  flags = spin_lock_irqsave(NULL);
-  value = *ptr;
-  *ptr  = value | (1 << bitno);
-  spin_unlock_irqrestore(NULL, flags);
+  value = (bt_atomic_t)atomic_fetch_or((FAR atomic_char *)ptr, (1 << bitno));
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.c:58]: let's typedef bt_atomic_t to some atomic type to avoid the cast

Comment (hujun260) [reply wireless/bluetooth/bt_atomic.c]: ok

File: wireless/bluetooth/bt_atomic.c [hunk 3]
```diff
@@ -35,78 +35,54 @@
 
 bt_atomic_t bt_atomic_incr(FAR bt_atomic_t *ptr)
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.c:36]: change to macro

Comment (hujun260) [reply wireless/bluetooth/bt_atomic.c]: done

**Review by anchao — APPROVED** [review]

File: wireless/bluetooth/bt_atomic.h [hunk 4]
```diff
@@ -37,26 +37,20 @@
...
- ****************************************************************************/
-
-bt_atomic_t bt_atomic_incr(FAR bt_atomic_t *ptr);
-bt_atomic_t bt_atomic_decr(FAR bt_atomic_t *ptr);
-bt_atomic_t bt_atomic_setbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-bt_atomic_t bt_atomic_clrbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-
-bool bt_atomic_testsetbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-bool bt_atomic_testclrbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
+typedef atomic_char bt_atomic_t;
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.h:54]: algin with https://github.com/apache/nuttx/pull/14827:
```suggestion
typedef atomic_t bt_atomic_t;
```

File: wireless/bluetooth/bt_atomic.h [hunk 5]
```diff
@@ -37,26 +37,20 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define bt_atomic_set(ptr, value)     (*(ptr) = (value))
-#define bt_atomic_get(ptr)            (*(ptr))
-#define bt_atomic_testbit(ptr, bitno) ((*(ptr) & (1 << (bitno))) != 0)
+#define bt_atomic_set(ptr, value)        atomic_store((ptr), (value));
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.h:40]:
```suggestion
#define bt_atomic_set(ptr, value)        atomic_store(ptr, value)
```

File: wireless/bluetooth/bt_atomic.h [hunk 6]
```diff
@@ -37,26 +37,20 @@
...
+#define bt_atomic_get(ptr)               atomic_load(ptr)
+#define bt_atomic_testbit(ptr, bitno)    ((atomic_load(ptr) & (1 << (bitno))) != 0)
+#define bt_atomic_incr(ptr)              atomic_fetch_add((ptr), 1)
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.h:43]:
```suggestion
#define bt_atomic_incr(ptr)              atomic_fetch_add(ptr, 1)
```

File: wireless/bluetooth/bt_atomic.h [hunk 7]
```diff
@@ -37,26 +37,20 @@
...
+#define bt_atomic_decr(ptr)              atomic_fetch_sub((ptr), 1)
+#define bt_atomic_setbit(ptr, bitno)     atomic_fetch_or((ptr), ((1 << (bitno))))
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.h:45]:
```suggestion
#define bt_atomic_setbit(ptr, bitno)     atomic_fetch_or(ptr, 1 << (bitno))
```

File: wireless/bluetooth/bt_atomic.h [hunk 8]
```diff
@@ -37,26 +37,20 @@
...
+#define bt_atomic_clrbit(ptr, bitno)     atomic_fetch_and((ptr), (~(1 << (bitno))))
```
Comment (xiaoxiang781216) [inline wireless/bluetooth/bt_atomic.h:46]:
```suggestion
#define bt_atomic_clrbit(ptr, bitno)     atomic_fetch_and(ptr, ~(1 << (bitno)))
```

**Comment by hujun260** [comment]
done

File: wireless/bluetooth/bt_atomic.h (hunk 8 above)
Comment (hujun260) [reply wireless/bluetooth/bt_atomic.h]: done

File: wireless/bluetooth/bt_atomic.h (hunk 7 above)
Comment (hujun260) [reply wireless/bluetooth/bt_atomic.h]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**